aten/src/ATen/native/cuda/LinearAlgebra.cu

```diff
@@ -400,15 +400,19 @@ Tensor dot_cuda(const Tensor& self, const Tensor& other) {
     TORCH_CUDABLAS_CHECK(
         cublasSetPointerMode(handle, CUBLAS_POINTER_MODE_DEVICE));
```

Comment:
You don't have to do it in this PR, but consider making an RAII guard for this idiom. Then you don't have to do a catch-all try-catch block.

Reply:
Good idea, I'll make a follow-up PR for that.